osbuild/util/ostree.py

```diff
@@ -225,16 +231,17 @@ def parse_deployment_option(root, deployment):
             raise ValueError("Could not find deployment")
         if len(filenames) > 1:
             raise ValueError("More than one deployment found")
-        
-        path = filenames[0].split("/")
-        osname = path[-3]
-        ref, serial = path[-1].split(".")
-        return osname, ref, serial
+
+        f = re.search("/ostree/deploy/(.*)/deploy/(.*)\\.([0-9])", filenames[0])
+        osname = f.group(1)
+        commit = f.group(2)
+        serial = f.group(3)
+        return osname, commit, serial
 
     osname = deployment["osname"]
-    ref = deployment["ref"]
+    commit = deployment["ref"]
     serial = deployment.get("serial", 0)
-    return osname, ref, serial
+    return osname, commit, serial
```

Comment:
i still say we use `ref` as the variable here since we are just returning back what the user/manifest provided.


```
ref = deployment["ref"]
serial = deployment.get("serial", 0)
return osname, ref, serial
```

Reply:
So make the auto-detect path return `commit` and the user input path return `ref`?

Reply:
yes, to the caller it's no different, but nice here to make things more explicit inside this function IMO